File: src/main/kotlin/com/nazarethlabs/notes/service/FavoriteNoteService.kt
package com.nazarethlabs.notes.service

import com.nazarethlabs.notes.dto.Note
import com.nazarethlabs.notes.repository.NoteStorageRepository

class FavoriteNoteService {
    fun toggleFavorite(note: Note) {
        NoteStorageRepository
            .getInstance()
            .toggleFavorite(note.id)
    }
}

File: src/core/service/FavoriteNoteService.ts
import { Note } from '../dto/Note';
import { NoteRepository } from '../repository/NoteRepository';
import { DateHelper } from '../helper/DateHelper';

export class FavoriteNoteService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  toggleFavorite(note: Note): void {
    const updatedNote: Note = {
      ...note,
      isFavorite: !note.isFavorite,
      updatedAt: DateHelper.nowMs(),
    };
    this.repository.save(updatedNote);
  }
}
